src/js/components/NetworkErrorNotice.tsx

```diff
@@ -15,21 +16,39 @@ const MAX_BACKOFF = 128
 
 export default function NetworkErrorNotice({error}: Props) {
```

Comment:
I'm having difficulty understanding the contents of this component now.  What are the use cases that each 'useEffect' call are intended for?